packages/xo-web/src/xo-app/vm/tab-advanced.js

```diff
@@ -749,6 +749,19 @@ export default class TabAdvanced extends Component {
                     />
                   </td>
                 </tr>
+                <tr>
+                  <th>{_('protectFromShutdown')}</th>
+                  <td>
+                    <Toggle
+                      value={'shutdown' in vm.blockedOperations}
+                      onChange={blockShutdown =>
+                        editVm(vm, {
+                          blockedOperations: { shutdown: blockShutdown },
+                        })
```

Comment:
This should probably block all the operations that would make the VM unavailable:
- `shutdown`
- `hard_shutdown`
- `reboot`
- `hard_reboot`
- `pause`
- `suspend`

→ http://xapi-project.github.io/xen-api/classes/vm.html

Reply:
Thanks for that. The clears up a few of the details for me.

Re: other commit relating to deletion prevention... I referenced the wrong commit. It should be f942127

Reply:
I think we don't need to make a "Do you want to do it anyway" modal for now, if that's what you mean. If you block the shutdown operations and then try to shutdown the VM, it should trigger an error message that is clear enough to understand what's going on. We might add it in the future but for now, the important part is just to be able to block the operations.

Reply:
Addressed with later commit.